ocaml/xapi/xapi_vm_lifecycle.ml

```diff
@@ -371,6 +371,15 @@ let check_operation_error ~__context ~vmr ~vmgmr ~ref ~clone_suspended_vm_enable
       | _ -> None
     ) in
 
+  (* CP-18860: can't set dynamic mem range when using nested virt *)
+  let current_error = check current_error (fun () ->
```

Comment:
I think that you are not catching all the cases here. A better place for this constraints seems to be `Xapi_vm_memory_constraints.assert_valid_for_current_context`. This is where we encode a similar constraint for dom0, and for the case that ballooning feature is disabled.